lib/msal-angular/src/msal-guard.service.ts

```diff
@@ -8,89 +8,46 @@ import {MSAL_CONFIG, MsalService} from "./msal.service";
 import 'rxjs/add/operator/filter';
 import 'rxjs/add/operator/pairwise';
 import {Location, PlatformLocation} from "@angular/common";
-import {MsalConfig} from "./msal-config";
 import {BroadcastService} from "./broadcast.service";
-import {Constants} from "msal";
-import {MSALError} from "./MSALError";
-import {AuthenticationResult} from "./AuthenticationResult";
+import { Configuration, AuthResponse, AuthError } from "msal";
 
 @Injectable()
 export class MsalGuard implements CanActivate {
 
-    constructor(@Inject(MSAL_CONFIG) private config: MsalConfig, private authService: MsalService, private router: Router, private activatedRoute: ActivatedRoute, private location: Location, private platformLocation: PlatformLocation, private broadcastService: BroadcastService) {
+    constructor(@Inject(MSAL_CONFIG) private msalConfig: Configuration, private authService: MsalService, private router: Router, private activatedRoute: ActivatedRoute, private location: Location, private platformLocation: PlatformLocation, private broadcastService: BroadcastService) {
     }
 
     canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean | Promise<boolean> {
         this.authService.getLogger().verbose("location change event from old url to new url");
 
-        this.authService.updateDataFromCache([this.config.clientID]);
-        if (!this.authService._oauthData.isAuthenticated && this.isObjectEmpty(this.authService._oauthData.idToken)) {
-            if (state.url) {
-
-                if (!this.authService._renewActive && !this.authService.loginInProgress()) {
-
-                    var loginStartPage = this.getBaseUrl() + state.url;
-                    if (loginStartPage !== null) {
-                        this.authService.getCacheStorage().setItem(Constants.angularLoginRequest, loginStartPage);
-                    }
-                    if (this.config.popUp) {
-                        return this.authService.loginPopup(this.config.consentScopes, this.config.extraQueryParameters)
-                            .then(function (token) {
-                               return true;
-                            }, function (error) {
-                                return false;
-                            });
-                    }
-                    else {
-                        this.authService.loginRedirect(this.config.consentScopes, this.config.extraQueryParameters);
-                    }
-                }
+        if (!this.authService.getAccount()) {
+            if (this.msalConfig.framework.popUp) {
```

Comment:
minor thing, but comments would be good here.

Reply:
Yeah, I will definitely comment this library more. Think I'll do that in a separate PR.